classes/log/EmailLogEntry.php

```diff
@@ -11,230 +11,143 @@
     {
-        $this->setData('assocId', $assocId);
+        return Attribute::make(
+            get: function () {
+                $sender = $this->senderId
+                    ? Repo::user()->get($this->senderId, true)
+                    : null;
+                return $sender ? $sender->getFullName() : '';
+            },
+        )->shouldCache();
     }
 
     /**
-     * Return the full name of the sender (not necessarily the same as the from address).
+     * Return the email address of sender.
      *
      * @return string
      */
-    public function getSenderFullName()
+    protected function senderEmail(): Attribute
     {
-        $senderFullName = $this->getData('senderFullName');
-
-        if ($senderFullName) {
-            return $senderFullName;
-        }
+        return Attribute::make(
+            get: function () {
+                $email = Repo::user()->get($this->senderId, true)->getEmail();
 
-        $sender = $this->getSenderId()
-            ? Repo::user()->get($this->getSenderId(), true)
-            : null;
-
-        return $sender ? $sender->getFullName() : '';
+                return $email ?:'';
+            },
+        )->shouldCache();
     }
 
+
     /**
-     * Return the email address of sender.
+     * Returns the subject of the message with a prefix explaining the event type
      *
-     * @return string
+     * @return string Prefixed subject
      */
-    public function getSenderEmail()
+    protected function prefixedSubject(): Attribute
     {
-        $senderEmail = & $this->getData('senderEmail');
-
-        if (!isset($senderEmail)) {
-            $senderEmail = Repo::user()->get($this->getSenderId(), true)->getEmail();
-        }
-
-        return ($senderEmail ? $senderEmail : '');
+        return Attribute::make(
+            get: fn() => __('submission.event.subjectPrefix') . ' ' . $this->subject
+        )->shouldCache();
     }
 
-
     //
-    // Email data
+    // Scopes
     //
-
-    public function getFrom()
-    {
-        return $this->getData('from');
-    }
-
-    public function setFrom($from)
-    {
-        $this->setData('from', $from);
-    }
-
-    public function getRecipients()
-    {
-        return $this->getData('recipients');
-    }
-
-    public function setRecipients($recipients)
-    {
-        $this->setData('recipients', $recipients);
-    }
-
-    public function getCcs()
-    {
-        return $this->getData('ccs');
-    }
-
-    public function setCcs($ccs)
-    {
-        $this->setData('ccs', $ccs);
-    }
-
-    public function getBccs()
-    {
-        return $this->getData('bccs');
-    }
-
-    public function setBccs($bccs)
-    {
-        $this->setData('bccs', $bccs);
-    }
-
-    public function getSubject()
-    {
-        return $this->getData('subject');
-    }
-
-    public function setSubject($subject)
+    public function scopeWithAssocId(Builder $query, int $submissionId = null): Builder
```

Comment:
Use `$assocId` rather than `$submissionId` -- we don't know for sure here that an assoc ID will be a submission ID. Also, I'm not 100% sure this won't have knock-on effects, but I'd suggest using a strict `int` rather than `?int`. Elsewhere, `null`s in filter functions are intended to mean "do not filter on this column", and allowing `null` here to get silently cast into an `int` might lead to confusion later. (Just FYI, when we do explicitly want to filter by a `null` value, we typically use functions that take arrays -- in this case it would be `scopeWithAssocIds` -- and accept a `[null]` filter value to explicitly mean "I want to select by null values".)

Reply:
(Further to this -- you'll note below that the `deleteByAssoc` function requires `int` parameters, not `?int`. It would be risky in that case to allow `?int` parameters and either cast them to zero, or [use conditional queries](https://github.com/pkp/pkp-lib/pull/10150/files#diff-fb677884b3042d12be73a0449956cc5ea712654c3ba980d1242728b288db663eR141), because user-supplied inputs might pass `null` through and suddenly affect a whole range of entries rather than an intended one. Let's try to stick consistent with that convention, which is safer.)